internal/config/policy.go

```
@@ -47,25 +49,45 @@ type Policy struct {
 	// TLSCustomCA defines the  root certificate to use with a given
 	// route when verifying server certificates.
 	TLSCustomCA string `mapstructure:"tls_custom_ca" yaml:"tls_custom_ca"`
+	RootCAs     *x509.CertPool
+
+	// Contains the x.509 client certificate to to present to the downstream
+	// host.
+	TLSClientCert     string `mapstructure:"tls_client_cert" yaml:"tls_client_cert"`
+	TLSClientKey      string `mapstructure:"tls_client_key" yaml:"tls_client_key"`
+	ClientCertificate *tls.Certificate
 }
 
 // Validate checks the validity of a policy.
 func (p *Policy) Validate() error {
 	var err error
 	p.Source, err = urlutil.ParseAndValidateURL(p.From)
 	if err != nil {
-		return fmt.Errorf("internal/config: bad source url %s", err)
+		return fmt.Errorf("internal/config: policy bad source url %s", err)
 	}
 
 	p.Destination, err = urlutil.ParseAndValidateURL(p.To)
 	if err != nil {
-		return fmt.Errorf("internal/config: bad destination url %s", err)
+		return fmt.Errorf("internal/config: policy bad destination url %s", err)
 	}
 
 	// Only allow public access if no other whitelists are in place
 	if p.AllowPublicUnauthenticatedAccess && (p.AllowedDomains != nil || p.AllowedGroups != nil || p.AllowedEmails != nil) {
-		return errors.New("internal/config: route marked as public but contains whitelists")
+		return fmt.Errorf("internal/config: policy route marked as public but contains whitelists")
+	}
+
+	if p.TLSClientCert != "" && p.TLSClientKey != "" {
```

Should probably throw an error if only one is configured.